src/components/remote-promo-sheet/check-fns/hasNonZeroTotalBalance.ts

```diff
@@ -1,16 +1,19 @@
+import { selectorFilterByUserChains, selectUserAssetsList } from '@/__swaps__/screens/Swap/resources/_selectors/assets';
+import { userAssetsFetchQuery } from '@/__swaps__/screens/Swap/resources/assets/userAssets';
 import store from '@/redux/store';
-import { fetchUserAssets } from '@/resources/assets/UserAssetsQuery';
 
 export const hasNonZeroTotalBalance = async (): Promise<boolean> => {
   const { accountAddress, nativeCurrency } = store.getState().settings;
 
-  const assets = await fetchUserAssets({
+  const userAssetsDictByChain = await userAssetsFetchQuery({
     address: accountAddress,
     currency: nativeCurrency,
-    connectedToHardhat: false,
+    testnetMode: false,
```

Comment:
again please use the getIsHardhatConnected util function so we don't have unexpected issues in e2e